lib/puppet/ffi/posix/functions.rb

```diff
@@ -0,0 +1,26 @@
+require 'puppet/ffi/posix'
+
+module Puppet::FFI::Posix
+  module Functions
+
+    extend FFI::Library
+
+    ffi_convention :stdcall
+
+    # https://man7.org/linux/man-pages/man3/getgrouplist.3.html
+    # https://www.gnu.org/software/libc/manual/html_node/Reading-Persona.html
+    # int getgrouplist (
+    #   const char *user,
+    #   gid_t group,
+    #   gid_t *groups,
+    #   int *ngroups
+    # );
+    begin
+      ffi_lib FFI::Library::LIBC
+      attach_function :getgrouplist, [:string, :uint, :pointer, :pointer], :int
+    rescue FFI::NotFoundError => e
+      puts e.message
```

Comment:
Will this be loud if it gets called on all puppet runs? Maybe we should just pass through without saying anything if the method does not exist.

Reply:
We could raise `LoadError` instead of puts, the error will be caught on https://github.com/puppetlabs/puppet/pull/8442/files#diff-d82f1437907787aa6ca7998d8fa0d360fd218e6decdd6a882c9aff1856a58af9R51 and we'll fallback to the old implementation.